src/test/txrebroadcast_tests.cpp

```diff
@@ -20,6 +20,24 @@
 
 #include <boost/test/unit_test.hpp>
 
+class TxRebroadcastHandlerTest : public TxRebroadcastHandler
+{
+public:
+    TxRebroadcastHandlerTest(CTxMemPool& mempool, ChainstateManager& chainman, const CChainParams& chainparams) :
+        TxRebroadcastHandler(mempool, chainman, chainparams){};
+
+    void UpdateAttempt(uint256 txhsh, int count)
+    {
+        auto attempt_time = GetTime<std::chrono::microseconds>() - 4h;
```

Comment:
In d579110263ad07386f1fbf796a3397a12340b6d5 "[test] Add unit test for rebroadcast attempt logic"

It seems a bit odd to me that this function is supposed to be a wrapper but it also implements a little bit of test logic in that it changes the recorded last attempt time. I think it would make more sense to have the main test logic calculate the timestamp change and make this function a pure wrapper (or perhaps remove the entire class).

Reply:
ok! removed the entire class and just made the test functions public with a comment indicating they are test only